components/video-player.tsx

```diff
@@ -41,8 +41,31 @@ type SizedEvent = {
   }
 };
 
-const VideoPlayer: React.FC<Props> = ({ playbackId, poster, onLoaded, onError }) => {
-  const videoRef = useRef<HTMLVideoElement | null>(null);
+export interface HTMLVideoElementWithPlyr extends HTMLVideoElement {
+  plyr: Plyr
+}
+
+const useCombinedRefs = function (...refs: (((instance: HTMLVideoElementWithPlyr | null) => void) | MutableRefObject<HTMLVideoElementWithPlyr | null> | null)[]) {
+  const targetRef = useRef(null);
+
+  useEffect(() => {
+    refs.forEach(ref => {
+      if (!ref) return;
+
+      if (typeof ref === 'function') {
+        ref(targetRef.current);
+      } else {
+        ref.current = targetRef.current;
+      }
+    });
+  }, [refs]);
+
+  return targetRef;
+};
+
+const VideoPlayer = forwardRef<HTMLVideoElementWithPlyr, Props>(({ playbackId, poster, onLoaded, onError }, ref) => {
+  const videoRef = useRef<HTMLVideoElementWithPlyr>(null);
+  const metaRef = useCombinedRefs(ref, videoRef);
```

Comment:
I'm not sure I'm following what's happening here with the combined refs bit, mind walking me through it?

Reply:
Yes!  So with how refs work in React, you can only associate one ref to a RefObject.  This is currently being used within the `VideoPlayer` to control setting the source to the player itself.  However, I wanted to also use it on the embed because I wanted to be able to control the playback state when the user opens/closes the overlay.

Another route I could take is to implement `useImparativeHandles` which will expose a `play`, `pause` and `isPlaying` functions as an alternative.

Do you have a preference?

Reply:
To summarize: this pattern is used when you need to forward the ref *and* use the ref internally in the component.

Reply:
Yea, that ^ 😉

Reply:
@dylanjha are you okay with `useCombinedRefs`?  Or would you prefer I use `useImparativeHandles`?

Reply:
I don't feel strongly, I think what you have is how I would have done it